tools/pubsys/src/main.rs

```diff
@@ -151,6 +161,12 @@ mod error {
         #[snafu(display("Failed to build repo: {}", source))]
         Repo { source: crate::repo::Error },
 
+        #[snafu(display("Failed to validate repo(s): {}", source))]
+        ValidateRepo { source: crate::repo::Error },
+
+        #[snafu(display("{}", source))]
```

Comment:
I'm starting to realize this might not be what we want here. I can only use `#[snafu(context(false))]` for one error that has `crate::repo::Error` as the `source`. If I have more than one, then snafu gets confused when trying to convert the underlying `repo::Error` as the top level pubsys `Error`. 

The workaround is to have separate error types for each subcommand `validate_repo`, `check_repo_expiration` and `refresh_repo`. But then I would end up with a bunch of duplicate Errors so I'm reluctant to do that.

```
error[E0119]: conflicting implementations of trait `std::convert::From<repo::error::Error>` for type `error::Error`:
   --> pubsys/src/main.rs:148:21
    |
148 |     #[derive(Debug, Snafu)]
    |                     ^^^^^
    |                     |
    |                     first implementation here
    |                     conflicting implementation for `error::Error`
    |
    = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0119`.
error: could not compile `pubsys`.
```

Reply:
Makes sense - best to reverse it.